File: TourMateBE/Repositories/Models/TourBid.cs
﻿using System;
using System.Collections.Generic;

namespace Repositories.Models;

public partial class TourBid
{
    public int TourBidId { get; set; }

    public int AccountId { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public bool IsDeleted { get; set; }

    public int PlaceRequested { get; set; }

    public string Status { get; set; } = null!;

    public string Content { get; set; } = null!;

    public float? MaxPrice { get; set; }

    public virtual Account Account { get; set; } = null!;

    public virtual ICollection<Bid> Bids { get; set; } = new List<Bid>();

    public virtual ActiveArea PlaceRequestedNavigation { get; set; } = null!;

    public virtual ICollection<TourBidComment> TourBidComments { get; set; } = new List<TourBidComment>();

    public virtual ICollection<UserLikeBid> UserLikeBids { get; set; } = new List<UserLikeBid>();
}

File: tourmatefe/types/tour-bid.ts
import { Account } from "./account"
import { ActiveArea } from "./active-area"

export type TourBid = {
    tourBidId: number,
    accountId: number,
    createdAt: string,
    updatedAt?: string,
    isDeleted: boolean,
    placeRequested: number,
    status: string,
    content: string,
    maxPrice?: number,
    account?: Account,
    placeRequestedNavigation?: ActiveArea,
}
export interface TourBidListResult {
  tourBidId: number;
  accountId: number;
  customerName: string;
  createdAt: string; // or Date if you'll convert it
  placeRequested: number;
  placeRequestedName: string;
  status: string;
  content: string;
  maxPrice?: number;
  likeCount: number;
  isLiked: boolean;
  customerImg: string
}

export function statusToCode(s: string){
  s = s.trim().toLowerCase()
  switch(s){
    case 'hoạt động': return 1;
    case 'chấm dứt': return 2;
  }
  return -1
}

export function codeToStatus(s: number){
  switch(s){
    case 1: return 'Hoạt động';
    case 2: return 'Chấm dứt';
  }
  return ''
}